api/app/src/main/kotlin/packit/security/WebSecurityConfig.kt

```diff
@@ -26,8 +29,18 @@ import packit.security.oauth2.OAuth2UserService
 import packit.security.provider.JwtIssuer
 import packit.service.BasicUserDetailsService
 
+@ConditionalOnProperty(prefix = "auth", name = ["enabled"], havingValue = "true", matchIfMissing = true)
+@Configuration
 @EnableWebSecurity
 @EnableMethodSecurity
+class SecurityEnabledConfig {
```

Comment:
Would it make sense to apply the `ConditionalOnProperty` to the existing `WebSecurityConfig` Configuration class instead of creating a new one?

For example the `WebSecurityConfig.HttpSecurity.handleSecurePaths` method below has an `config.authEnabled` conditional, presumably that can go away?